environment/src/main/java/com/sequenceiq/environment/environment/service/DatahubService.java

```diff
@@ -0,0 +1,98 @@
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.exception.PollerException;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackViewV4Response;
+import com.sequenceiq.cloudbreak.common.exception.WebApplicationExceptionMessageExtractor;
+import com.sequenceiq.distrox.api.v1.distrox.endpoint.DistroXV1Endpoint;
+import com.sequenceiq.environment.environment.poller.DatahubPollerProvider;
+import com.sequenceiq.environment.exception.EnvironmentServiceException;
+
+@Service
+public class DatahubService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatahubService.class);
+
+    @Value("${env.stop.polling.attemptCount:90}")
+    private Integer attemptCount;
+
+    @Value("${env.stop.polling.sleep.time.seconds:5}")
+    private Integer sleepTime;
+
+    private final DistroXV1Endpoint distroXV1Endpoint;
+
+    private final DatahubPollerProvider datahubPollerProvider;
+
+    private final WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor;
+
+    public DatahubService(DistroXV1Endpoint distroXV1Endpoint, DatahubPollerProvider datahubPollerProvider,
+            WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor) {
+        this.distroXV1Endpoint = distroXV1Endpoint;
+        this.datahubPollerProvider = datahubPollerProvider;
+        this.webApplicationExceptionMessageExtractor = webApplicationExceptionMessageExtractor;
+    }
+
+    public void startAttachedDatahubClusters(Long envId, String envCrn) {
+        Collection<StackViewV4Response> attachedDistroXClusters = getAttachedDatahubClusters(envCrn);
+        List<String> datahubCrns = mapStacksToCrns(attachedDistroXClusters);
+        executeDatahubOperationAndStartPolling(datahubCrns, distroXV1Endpoint::putStartByCrns,
+                datahubPollerProvider.startDatahubClustersPoller(datahubCrns, envId));
+
+    }
+
+    public void stopAttachedDatahubClusters(Long envId, String envCrn) {
+        Collection<StackViewV4Response> attachedDistroXClusters = getAttachedDatahubClusters(envCrn);
+        List<String> datahubCrns = mapStacksToCrns(attachedDistroXClusters);
+        executeDatahubOperationAndStartPolling(datahubCrns, distroXV1Endpoint::putStopByCrns,
+                datahubPollerProvider.stopDatahubClustersPoller(datahubCrns, envId));
+    }
+
+    private void executeDatahubOperationAndStartPolling(List<String> datahubCrns, Consumer<List<String>> distroxOperation, AttemptMaker<Void> attemptMaker) {
+        try {
+            distroxOperation.accept(datahubCrns);
+            Polling.stopAfterAttempt(attemptCount)
+                    .stopIfException(true)
+                    .waitPeriodly(sleepTime, TimeUnit.SECONDS)
+                    .run(attemptMaker);
+        } catch (PollerException e) {
+            if (e.getCause() != null && e.getCause() instanceof WebApplicationException) {
```

Comment:
otherwise you swallow exception. rethrow please! if no rethrow is needed please leave a comment in the code why is it normal.